raiden/network/rpc/client.py

```diff
@@ -631,7 +631,10 @@ def __init__(self, jsonrpc_client, registry_address, startgas=GAS_LIMIT,
 
     def manager_address_by_token(self, token_address):
         """ Return the channel manager address for the given token. """
-        return self.proxy.channelManagerByToken.call(token_address)
+        manager_address_encoded = self.proxy.channelManagerByToken.call(token_address)
+        if not manager_address_encoded or manager_address_encoded == '0x':
+            manager_address_encoded = self.add_token(token_address)
```

Comment:
This is what I consider proper behavior:

If you try to get a proxy to a non-existing channel manager None should be returned, this implies the token is not registered.
For the programmatic interfaces this result should be propagated, so in the rest API case that should result in a 404.
For the UIs the user should either be asked to register it (GUI), or informed that it is not registered and how to do it (CLI)